### ISSUE TITLE
Bark TTS with voice clone?

### ISSUE THREAD

**Issue by Ph0rk0z** [Cohee1207/TavernAI-extras#15]
Between expressions, memory and everything else, I think this would really kick immersion up.

https://github.com/serp-ai/bark-with-voice-clone

**Comment by Priestru** [comment]
I just opened issues to ask for Bark too!

**Comment by Ph0rk0z** [comment]
The cloning sort of sucks right now but the default voices aren't so bad.

**Comment by Ph0rk0z** [comment]
I used silero today.. It won't play the available voices for some reason. It does work though through that API server which I had to edit. Works ok, could use some quality of life improvements but it's aight. For instance to see if your voice file is done or if generation finished, etc.

I used the silero pip. Having 2 gpus, eager for tortoise or maybe even coqui. Once bark sorts itself out.. I think it will be awesome actually acting out your roleplay and not just reading like a robot.

**Comment by FerLuisxd** [comment]
Would love to see Bark TTS here!

**Comment by Ph0rk0z** [comment]
There is also a new bark I am trying https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer 

Cloning might work in that one.

**Comment by gitmylo** [comment]
didn't expect to see my repo mentioned here lol

(If anyone needs help implementing the voice cloning quantizer, I'll gladly help. Note that there currently isn't a multilingual model, but once enough languages are made by the community, I can train a multilingual model by combining the datasets.)

**Comment by Ph0rk0z** [comment]
It's the best clones I've gotten with so little data. If only bark could keep it up from sentence to sentence.

**Comment by gitmylo** [comment]
Yeah, if you want better results at somewhat higher inference time, combine it with RVC, included in my [audio-webui](/gitmylo/audio-webui), just generate with bark, then click "Send to RVC". Make sure you have a trained RVC model, I'm currently working on porting the training code to the webui for an easier interface. Button to estimate epochs, stuff like that.

**Comment by Ph0rk0z** [comment]
I'm waiting on you to put in RVC training and making some tortoise models in the mean time. It doesn't do too bad and trains in 5hrs.

Just generation on all of these is a bit slow for real time text. Hopefully one day that gets solved because it takes a bit even on a 3090.

**Comment by gitmylo** [comment]
5 hours?

With bark + RVC, the training of RVC takes like half an hour total, you won't have to train for long, and the results will probably be better than tortoise. (example below)


https://github.com/SillyTavern/SillyTavern-extras/assets/36931363/362d6066-6e0a-4c52-a9e3-3ca12773931e

**Comment by gitmylo** [comment]
For cloning with bark alone, no training is required, just a short audio clip. This clip is then sent to bark as a history prompt, and the rest will be generated from there.

There could be an option to create a voice from an audio clip. Which is then used for generations with bark. Creating a speaker is faster than generating with bark itsself, so it should be fine being built-in.

**Comment by pyrater** [comment]
bark is incoming with the addition of coqui. It is working on my dev install. https://github.com/SillyTavern/SillyTavern-extras/tree/neo and  https://github.com/SillyTavern/SillyTavern/pull/775

**Comment by Ph0rk0z** [comment]
Thanks! I have bark working in textgen but I prefer all the extra features ST for RP.  I'm a lucky one and have a whole 24g GPU to dedicate to SD + TTS. I will also try to train a vits tho because why not. Bark consistency on longer gens pretty much isn't.
